### ISSUE TITLE
elVeh is not a numeric value

### ISSUE THREAD

**Issue by Cellivar** [itchannel/fordpass-ha#173]
Apparently the elVeh value is coming in as a non-numeric value. I had to do some shenanigans to perform math on the value in my influxbd server. 

![Screenshot_20220424-031110_Home Assistant](https://user-images.githubusercontent.com/1441553/164971569-55b893c7-7759-44b2-834b-3a76f2e7e46a.jpg)
 
Possibly related, the timestamp values all appear to be raw text instead of a timestamp value that can be used for math operations. I haven't done more digging to see if there are other datatype mismatches floating around.

**Comment by itchannel** [comment]
Thanks for the feedback. Currently all the values come back from the Ford API as string values but I will look into casting them into their relevant format.

**Comment by DrFrankReade** [comment]
Hello, I'll second this report as a bug. Unfortunately I don't know python enough to cast it as a float or int, although I see the conversion to miles, and it's float there obviously to do the math... 

```
            elif self.sensor == "elVeh":
                if self.coordinator.data["elVehDTE"] != None:
                    if self.options[CONF_DISTANCE_UNIT] != None:
                        if self.options[CONF_DISTANCE_UNIT] == "mi":
                            return round(
                                float(self.coordinator.data["elVehDTE"]["value"])
                                / 1.60934
                            )
                        else:
                            return self.coordinator.data["elVehDTE"]["value"]
                    else:
                        return self.coordinator.data["elVehDTE"]["value"]
                else:
                    return "Unsupported"
```
Other values like odometer and number of messages are coming in as numeric.

**Comment by itchannel** [comment]
I don't have an electric car but is the battery fill level a whole number or a decimal?

Once I know I'll fix these two bugs 👍

**Comment by DrFrankReade** [comment]
Hello,

Battery fill level is a whole integer, no decimal.

ElVehDTE is distance to empty and it has one decimal place - 123.4


Thank you for sharing your work with the community.

Andy Baker

**Comment by itchannel** [comment]
elVeh fixed in 1.48
